service/storage_proxy.cc

```diff
@@ -3853,13 +3853,15 @@ class speculating_read_executor : public abstract_read_executor {
 };
 
 db::read_repair_decision storage_proxy::new_read_repair_decision(const schema& s) {
-    double chance = _read_repair_chance(_urandom);
-    if (s.read_repair_chance() > chance) {
-        return db::read_repair_decision::GLOBAL;
-    }
+    if (s.dc_local_read_repair_chance() > 0 || s.read_repair_chance() > 0) {
+        double chance = _read_repair_chance(_urandom);
```

Comment:
Nit: I would add a comment on this line of code explaining that it's expensive on ARM and thus should be carefully used.